hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/store/contracts/precompile/ExchangeRatePrecompiledContract.java

```diff
@@ -82,6 +85,7 @@ public long gasRequirement(Bytes bytes) {
     @Override
     public Bytes compute(final Bytes input, final MessageFrame frame) {
         try {
+            validateTrue(input.size() >= 4, INVALID_TRANSACTION_BODY);
```

Comment:
Hey @tinker-michaelj 
In all 3 precompiled contracts in the mono, there are catch blocks that will catch `InvalidTransactionException` and ultimately return `PrecompileContractResult `.
For reference :
`HtsPrecompiledContract` line 240
`PrngSystemPrecompiledContract` line 144
for `ExchangeRatePrecompiledContract` we go into `PrecompiledContract` default logic which still returns `PrecompiledContract.PrecompileContractResult.halt`

Reply:
Not related to the issue but we only do `frame.setExceptionalHaltReason(Optional.of(haltReason));`
in `HtsPrecompiledContract` and not the other 2. Is there a reason for that?

Reply:
Hmm you're right @0xivanov ... I definitely prefer not to introduce `validateTrue()` anywhere in mod-service system contracts; but we already have to refactor everything there, so it can be cleaned up later.

I suspect only `HtsPrecompiledContract` sets a halt reason explicitly because the other two precompiles will return `null` if they fail; and then the inherited `computePrecompile()` implementation [here](https://github.com/hyperledger/besu/blob/main/evm/src/main/java/org/hyperledger/besu/evm/precompile/PrecompiledContract.java#L61) sets the halt reason automatically.